diff --git a/src/kcolorschememanager.cpp b/src/kcolorschememanager.cpp
index ce59ec3..b240210 100644
--- a/src/kcolorschememanager.cpp
+++ b/src/kcolorschememanager.cpp
@@ -106,14 +106,14 @@ KColorSchemeManager::KColorSchemeManager(GuardApplicationConstructor, QGuiApplic
     init();
 }
 
-#if KCOLORSCHEME_BUILD_DEPRECATED_SINCE(6, 6)
+//#if KCOLORSCHEME_BUILD_DEPRECATED_SINCE(6, 6)
 KColorSchemeManager::KColorSchemeManager(QObject *parent)
     : QObject(parent)
     , d(new KColorSchemeManagerPrivate())
 {
     init();
 }
-#endif
+//#endif
 
 KColorSchemeManager::~KColorSchemeManager()
 {
diff --git a/src/kcolorschememanager.h b/src/kcolorschememanager.h
index d899485..6994574 100644
--- a/src/kcolorschememanager.h
+++ b/src/kcolorschememanager.h
@@ -54,10 +54,10 @@ class KCOLORSCHEME_EXPORT KColorSchemeManager : public QObject
 {
     Q_OBJECT
 public:
-#if KCOLORSCHEME_ENABLE_DEPRECATED_SINCE(6, 6)
+//#if KCOLORSCHEME_ENABLE_DEPRECATED_SINCE(6, 6)
     KCOLORSCHEME_DEPRECATED_VERSION(6, 6, "Use KColorSchemeManager::instance()")
     explicit KColorSchemeManager(QObject *parent = nullptr);
-#endif
+//#endif
 
     ~KColorSchemeManager() override;
 
