diff --git a/src/gui/platform/unix/qgenericunixservices.cpp b/src/gui/platform/unix/qgenericunixservices.cpp
index 708ce0d9c8d..952dee17c6e 100644
--- a/qtbase/src/gui/platform/unix/qgenericunixservices.cpp
+++ b/qtbase/src/gui/platform/unix/qgenericunixservices.cpp
@@ -153,12 +153,79 @@ static inline bool launch(const QString &launcher, const QUrl &url,
         process.setProgram(program);
         process.setArguments(args);
 
+        QProcessEnvironment env = QProcessEnvironment::systemEnvironment();
+        qDebug() << env.toStringList();
+
         if (!xdgActivationToken.isEmpty()) {
-            auto env = QProcessEnvironment::systemEnvironment();
             env.insert(u"XDG_ACTIVATION_TOKEN"_s, xdgActivationToken);
-            process.setEnvironment(env.toStringList());
         }
-        ok = process.startDetached(nullptr);
+
+        // If we are running into AppImage bundle, switch env var to the right values.
+        if (env.contains(QLatin1String("APPIMAGE_ORIGINAL_LD_LIBRARY_PATH")) &&
+            env.contains(QLatin1String("APPIMAGE_ORIGINAL_QT_PLUGIN_PATH"))  &&
+            env.contains(QLatin1String("APPIMAGE_ORIGINAL_XDG_DATA_DIRS"))   &&
+            env.contains(QLatin1String("APPIMAGE_ORIGINAL_LD_PRELOAD"))      &&
+            env.contains(QLatin1String("APPIMAGE_ORIGINAL_PATH")))
+        {
+            qDebug() << "Adjusting environment variables for AppImage bundle";
+
+            if (!env.value(QLatin1String("APPIMAGE_ORIGINAL_LD_LIBRARY_PATH")).isEmpty())
+            {
+                env.insert(QLatin1String("LD_LIBRARY_PATH"),
+                           env.value(QLatin1String("APPIMAGE_ORIGINAL_LD_LIBRARY_PATH")));
+            }
+            else
+            {
+                env.remove(QLatin1String("LD_LIBRARY_PATH"));
+            }
+
+            if (!env.value(QLatin1String("APPIMAGE_ORIGINAL_QT_PLUGIN_PATH")).isEmpty())
+            {
+                env.insert(QLatin1String("QT_PLUGIN_PATH"),
+                           env.value(QLatin1String("APPIMAGE_ORIGINAL_QT_PLUGIN_PATH")));
+            }
+            else
+            {
+                env.remove(QLatin1String("QT_PLUGIN_PATH"));
+            }
+
+            if (!env.value(QLatin1String("APPIMAGE_ORIGINAL_XDG_DATA_DIRS")).isEmpty())
+            {
+                env.insert(QLatin1String("XDG_DATA_DIRS"),
+                           env.value(QLatin1String("APPIMAGE_ORIGINAL_XDG_DATA_DIRS")));
+            }
+            else
+            {
+                env.remove(QLatin1String("XDG_DATA_DIRS"));
+            }
+
+            if (!env.value(QLatin1String("APPIMAGE_ORIGINAL_LD_PRELOAD")).isEmpty())
+            {
+                env.insert(QLatin1String("LD_PRELOAD"),
+                           env.value(QLatin1String("APPIMAGE_ORIGINAL_LD_PRELOAD")));
+            }
+            else
+            {
+                env.remove(QLatin1String("LD_PRELOAD"));
+            }
+
+            if (!env.value(QLatin1String("APPIMAGE_ORIGINAL_PATH")).isEmpty())
+            {
+                env.insert(QLatin1String("PATH"),
+                           env.value(QLatin1String("APPIMAGE_ORIGINAL_PATH")));
+            }
+            else
+            {
+                env.remove(QLatin1String("PATH"));
+            }
+
+            qDebug() << env.toStringList();
+        }
+
+        process.setProcessEnvironment(env);
+        process.setProgram(program);
+        process.setArguments(args);
+        ok = process.startDetached();
     }
 #  endif
     if (!ok)
